aiida_pseudo/groups/mixins/cutoffs.py

```diff
@@ -85,13 +85,13 @@ def validate_stringency(self, stringency: str) -> None:
         """Validate a cutoff stringency.
 
         :param stringency: the cutoff stringency to validate.
-        :raises ValueError: if stringency is None or the family does not define cutoffs for the specified stringency.
+        :raises ValueError: if the family does not define cutoffs for the specified stringency.
         """
-        if stringency is None:
```

Comment:
I'm not sure why this had to raise, since `stringency` is not an optional input, and type hinting indicates it has to be a string.

Reply:
Well, because the following is allowed in Python:
```
In [1]: def test(a: str):
   ...:     print(a)
   ...: 

In [2]: test(None)
None

```
it is type _hinting_ and so not enforced. `None` is a valid option and if we don't accept it, we have to check it explicitly. 
The following:
```
In [4]: test()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-fbd55f77ab7c> in <module>
----> 1 test()

TypeError: test() missing 1 required positional argument: 'a'
```
raises though.
This again comes back to our discussion recently on the protocols and the fact that `None` for an argument and the absence of an argument are not the same thing.

Reply:
Sure, I understand that someone could pass `None` and it would raise with a slightly more clear error message if we catch this, but this could also be said for the `unit` in `validate_cutoffs_unit`, or basically almost any input of any method. Someone could also pass a `tuple`, `list` etc. I just didn't understand why we had to catch this user error specifically, since it would still raise because `None` wouldn't be in the list of defined stringencies.

However, now that inputting `None` checks if there is a _default_ stringency set, the point is moot. 😄